src/components/textarea/textarea.test.jsx

```diff
@@ -0,0 +1,227 @@
+      rerender(<Textarea {...requiredProps} ref={ref} value="a" />);
+      expect(ref.current.dirty).toBe(false);
+    });
+  });
+
+  describe('id API', () => {
+    it('sets the ID', () => {
+      render(<Textarea {...requiredProps} id="test-id" />);
+      expect(screen.getByLabelText('the label')).toHaveAttribute('id', 'test-id');
+    });
+
+    it('is set on the ref', () => {
+      const ref = createRef();
+      render(<Textarea {...requiredProps} id="unique-id" ref={ref} />);
+      expect(ref.current.id).toBe('unique-id');
+    });
+  });
+
+  describe('label API', () => {
+    it('can be set', () => {
+      render(<Textarea {...requiredProps} label="Unique label" />);
+      expect(screen.getByLabelText('Unique label')).toBeInTheDocument();
+    });
+  });
+
+  describe('name API', () => {
+    it('sets the name attribute', () => {
+      render(<Textarea {...requiredProps} name="test-name" />);
+      expect(screen.getByLabelText('the label')).toHaveAttribute('name', 'test-name');
+    });
+
+    it('is set on the ref', () => {
+      const ref = createRef();
+      render(<Textarea {...requiredProps} name="unique-name" ref={ref} />);
+      expect(ref.current.name).toBe('unique-name');
+    });
+  });
+
+  describe('onBlur API', () => {
+    it('sets the onblur handler', () => {
+      const onBlurSpy = jest.fn();
+      render(<Textarea {...requiredProps} onBlur={onBlurSpy} />);
+      fireEvent.blur(screen.getByLabelText('the label'));
+      expect(onBlurSpy.mock.calls.length).toEqual(1);
+    });
+  });
+
+  describe('onChange API', () => {
+    it('sets the onchange handler', () => {
+      const onChangeSpy = jest.fn();
+      render(<Textarea {...requiredProps} onChange={onChangeSpy} />);
+      fireEvent.change(screen.getByLabelText('the label'), { target: { value: 'new value' } });
+      expect(onChangeSpy.mock.calls.length).toEqual(1);
+    });
+  });
+
+  describe('placeholder API', () => {
+    it('sets the placeholder attribute', () => {
+      render(<Textarea {...requiredProps} placeholder="test-placeholder" />);
+      expect(screen.getByLabelText('the label')).toHaveAttribute('placeholder', 'test-placeholder');
+    });
+  });
+
+  describe('readOnly API', () => {
+    it('sets the readOnly attribute', () => {
+      render(<Textarea {...requiredProps} readOnly />);
+      expect(screen.getByLabelText('the label')).toHaveAttribute('readonly');
+    });
+  });
+
+  describe('ref API', () => {
+    it('can be set', () => {
+      const ref = React.createRef();
+      render(<Textarea {...requiredProps} ref={ref} value="test value" />);
+      expect(ref.current.value).toEqual('test value');
+    });
+  });
+
+  describe('required API', () => {
+    it('can be set', () => {
+      render(<Textarea {...requiredProps} required={true} />);
+      expect(screen.getByLabelText('the label')).toBeRequired();
+      expect(screen.getByLabelText('the label')).toBeInvalid();
```

Comment:
Let's add something about not seeing the validation message.

```suggestion
      expect(screen.getByLabelText('the label')).toBeInvalid();
      expect(screen.getByText(<le error text>)).not.toBeInTheDocument();
```